src/lib/utils/expression.ts

```diff
@@ -0,0 +1,25 @@
+import type { ExpressionSpecification } from 'maplibre-gl';
+
+import { COLORS, COLOR_ORDER } from './constants';
+
+/**
+ * Create a Maplibre GL JS match expression for a given field and categories.
+ *
+ * @param field – The field to match against, one of 'Agency Name', 'Category',
+ * or 'Funding Source'.
+ * @param categories – The categories to match against.
+ * @returns – A Maplibre GL JS match expression.
+ */
+export function createColorMatchExpression(
+	field: 'Agency Name' | 'Category' | 'Funding Source',
+	categories: string[]
+): ExpressionSpecification {
+	return [
+		'match',
+		['get', field],
+		categories[0],
+		COLOR_ORDER[0],
+		...categories.slice(1).flatMap((name, i) => [name, COLOR_ORDER[i + 1]]),
+		COLORS.EARTH
```

Comment:
Here's an example of the kind of goofy shit you do for TypeScript. In your original implementation, you had a cast of this whole array to `any` (presumably because the type error was inscrutable). In this case, I've been around the block enough to know that a `match` expression needs to statically be able to observe _at least two elements_ in the array. Thus, we "peel off" the first element (`categories[0]`) and then construct the remaining elements (`...categories.slice(1).flatMap`). This provides enough information to pass the type check here.

TS is weird in this case. This is an awkward form of what's called "dependent typing", where we're actually asserting something about the runtime structure of the `categories` array (i.e., that it has at least two elements).